submissions/andrewklmn/friends-app/app.js

```diff
@@ -0,0 +1,334 @@
+  
+  friends.forEach((friend) => {
+    drawPerson(friend);
+  });
+  state.numberOfShowedFriends += friends.length
+  
+  setTimeout(() => {
+    if (container.scrollTop > 0) {
+      container.scrollTop += state.afterLoadScrollShift;
+    };
+  }, state.nextMoreFriendAutoloadDelay * 2);
+};
+
+const autoLoaderOnScroll = (state) => {
+  const {numberOfShowedFriends, friends, initialListLength, nextMoreFriendAutoloadDelay} = state;
+  
+  if (state.scrollDisabled) return;
+
+  if (container.scrollTop + container.clientHeight >= --container.scrollHeight) {
+    if(numberOfShowedFriends < friends.length) {
+      state.scrollDisabled = true;
+      container.scrollTop--;
+      preloader.classList.remove('hidden');
+
+      const additionalFriendList = sortList(filterList(state))
+        .slice( numberOfShowedFriends, numberOfShowedFriends + initialListLength);
+
+      requestAnimationFrame(() => drawMoreFriends(additionalFriendList) );
+
+      setTimeout(()=>{    
+        state.scrollDisabled = false;
+        preloader.classList.add('hidden');
+      }, nextMoreFriendAutoloadDelay);
+    };
+  }
+};
+
+const sorterKeyChangeHandler = (target, state) => {
+  const {sorter} = state;
+  if (target.value === '') {    
+    sortOrder.innerHTML = '';
+    sorter.keyName = '';
+    sorter.order = '';
+  } else {
+    if(sortOrder.innerHTML === ''){
+      sortOrder.innerHTML = sorter.orderSymbols.ASC;
+      sorter.order = 'ASC';
+    };    
+    sorter.keyName = target.value; 
+  };
+  redrawFriends(state);
+}
+
+const sorterOrderClickHandler = (target, state)=>{
+  const {sorter} = state;
+
+  if(sortField.value === ''){
+    target.innerHTML = '';
+    sorter.order = '';
+    return;
+  };
+  if (sorter.order === 'ASC') {
+    target.innerHTML = sorter.orderSymbols.DESC;
+    sorter.order = 'DESC';
+  } else {
+    target.innerHTML = sorter.orderSymbols.ASC;
+    sorter.order = 'ASC';
+  };
+
+  redrawFriends(state);
+}
+
+const filterTextChangeHandler = (target, state, nameOfFilterProp) => {
+  const {filters} = state;
+  if (filters[nameOfFilterProp] === target.value) {
+    return true;
+  }
+  filters[nameOfFilterProp] = target.value;
+  redrawFriends(state);
+}
+
+const MIN_AGE_RANGE_INDEX = 0; 
+const MAX_AGE_RANGE_INDEX = 1;
+
+const filterAgeChangeHandler = (target, state, ageRangeIndex)=>{
+  state.filters.ageRange[ageRangeIndex] = target.value;
+  redrawFriends(state);
+}
+
+document.addEventListener('DOMContentLoaded',() => {
+  initApp(state);
+  
+  sortField.addEventListener('change', ({target}) => sorterKeyChangeHandler(target, state));
+  sortOrder.addEventListener('click', ({target}) => sorterOrderClickHandler(target, state));
+  
+  resetButton.addEventListener('click', () => resetButtonClickHandler(state));
+  
+  searchField.addEventListener('keyup',({target}) => filterTextChangeHandler(target, state, 'partOfName'));
+  filterGender.addEventListener('change', ({target}) => filterTextChangeHandler(target, state, 'gender'));
```

Comment:
Also, I'd recommend combining these event listeners into one. You can use event delegation and check target.